core/src/raw/http_util/header.rs

```diff
@@ -189,6 +191,21 @@ pub fn parse_into_metadata(path: &str, headers: &HeaderMap) -> Result<Metadata>
     Ok(m)
 }
 
+/// Parse prefixed headers and return a map with the prefix of each header removed.
+pub fn parse_prefixed_headers(headers: &HeaderMap, prefix: &str) -> HashMap<String, String> {
+    headers
+        .iter()
+        .filter_map(|(name, value)| {
+            name.as_str().strip_prefix(prefix).and_then(|stripped_key| {
+                value
+                    .to_str()
```

Comment:
I didn't use the `parse_header_to_str` as it was done in the `s3` user_meta implementation. I see that the `parse_header_to_str` was only useful in this case for the error handling of `value.to_str()`, but that error handling was being thrown away 
![image](https://github.com/user-attachments/assets/1217902a-10f6-41d1-b8f2-63d35a3cdc8b)

Changed to this for simplicity sake.